### ISSUE TITLE
Restarting Containers

### ISSUE THREAD

**Issue by morpheouso** [Sanderhuisman/docker_monitor#26]
Is it possible to restart containers with this like it was with HA-dockermon? EG:
`shell_command:
  restart_hass: 'curl http://127.0.0.1:8126/container/home-assistant/restart'`

**Comment by Kisty** [comment]
I am able to with a switch. In generates a switch for each docker container e.g. switch.docker_<your docker container>.

**Comment by matt8707** [comment]
Yes, but restarting the hass container from hass?

**Comment by Kisty** [comment]
@matt8707 What do you mean? Like restart another container other than home assistant?

**Comment by matt8707** [comment]
No, I mean stopping the home-assistant docker container from lovelace would cut the connection. You can't start the container again from the ui if you stopped home assistant. 

ha-dockermon has a `restart` command.

**Comment by Morphy99** [comment]
You can restart HA from the GUI? I guess if it hangs it could be an issue.

On the other side a restart is less cumbersome, you'd have to wait for the container to stop before starting again with the switch. Implementing that in a HA automation could be problematic.

**Comment by matt8707** [comment]
If i restart or stop from the ui hass doesn't load 30% of the time. Seems like I'm not alone with this issue.

> you'd have to wait for the container to stop before starting again with the switch. Implementing that in a HA automation could be problematic.

How would home assistant trigger an automation if home assistant isn't running?

**Comment by Morphy99** [comment]
I was referring to other containers not HA specifically in that sentence. For example I restart zigbee2mqtt on HA restart to properly detect the states of my devices.

I guess we agree that Dockermom could do with a restart function at least 😆

**Comment by matt8707** [comment]
I can replicate on a vanilla install of home assistant core in docker. Never had issues on hass.io so I'm not sure it's a config error. I mean it's not a huge deal, I can restart from shell.

**Comment by Morphy99** [comment]
> I can replicate on a vanilla install of home assistant core in docker. Never had issues on hass.io so I'm not sure it's a config error. I mean it's not a huge deal, I can restart from shell.

That's odd, maybe a docker config issue then? I have HA core docker running on RPi4 on Libreelec and restarts fine 99% of the time. Only time I can recall it not working is when I had a config error in my yaml I forgot to correct before restarting. What version of HA are you running? I'm one behind latest so maybe a bug in the most recent?

**Comment by matt8707** [comment]
Yeah I don't mean yaml config errors. Seems to be an Issue with synology nas somehow. Either way I just tried  ha-dockermon and it works as expected.
